src/db/mod.rs

```diff
@@ -0,0 +1,109 @@
+use diesel::{prelude::*, sqlite::SqliteConnection};
+
+pub mod models;
+pub mod schema;
+
+pub fn establish_connection() -> SqliteConnection {
+    let db = "./ichneos.db";
+    SqliteConnection::establish(db)
+        .unwrap_or_else(|_| panic!("Error connecting to {}", db))
+}
+
+pub fn save_activity(connection: &SqliteConnection,
```

Comment:
Can you put these `save_*` functions into a new file like `src/db/insertions` and call them with `pub mod insertions;`.